app/routes.test.js

```diff
@@ -197,6 +198,210 @@ describe('routes', () => {
+        .get('/organisations/org1/user1/enable')
+        .set('Cookie', [mockAuthorisedCookie])
+        .expect(200)
+        .then((res) => {
+          expect(res.text.includes('data-test-id="mock-confirmation-page"')).toEqual(true);
+          expect(res.text.includes('data-test-id="error-page-title"')).toEqual(false);
+        });
+    });
+  });
+
+  describe('POST /organisations/:organisationId/:userId/enable', () => {
+    afterEach(() => {
+      userStatusContext.postUserStatus.mockReset();
+    });
+
+    it('should return 403 forbidden if no csrf token is available', () => {
+      userStatusContext.postUserStatus = jest.fn()
+        .mockImplementation(() => Promise.resolve({ success: true }));
+
+      return request(setUpFakeApp())
+        .post('/organisations/org1/user2/enable')
+        .set('Cookie', [mockAuthorisedCookie])
+        .type('form')
+        .send({})
+        .expect(403);
+    });
+
+    it('should redirect to the login page if the user is not logged in', async () => {
+      viewUserContext.getViewUserContext = jest.fn()
+        .mockImplementation(() => Promise.resolve(mockAddUserData));
+      const { cookies, csrfToken } = await getCsrfTokenFromGet(setUpFakeApp(), '/organisations/org1/user2', mockAuthorisedCookie);
+
+      return request(setUpFakeApp())
+        .post('/organisations/org1/user2/enable')
+        .type('form')
+        .set('Cookie', [cookies])
+        .send({
+          _csrf: csrfToken,
+        })
+        .expect(302)
+        .then((res) => {
+          expect(res.redirect).toEqual(true);
+          expect(res.headers.location).toEqual('http://identity-server/login');
+        });
+    });
+
+    it('should show the error page indicating the user is not authorised if the user is logged in but not authorised', async () => {
+      viewUserContext.getViewUserContext = jest.fn()
+        .mockImplementation(() => Promise.resolve(mockAddUserData));
+      const { cookies, csrfToken } = await getCsrfTokenFromGet(setUpFakeApp(), '/organisations/org1/user2', mockAuthorisedCookie);
+
+      const mockUnauthorisedJwtPayload = JSON.stringify({
+        id: '88421113', name: 'Cool Dude',
+      });
+      const mockUnauthorisedCookie = `fakeToken=${mockUnauthorisedJwtPayload}`;
+
+      return request(setUpFakeApp())
+        .post('/organisations/org1/user2/enable')
+        .type('form')
+        .set('Cookie', [cookies, mockUnauthorisedCookie])
+        .send({ _csrf: csrfToken })
+        .expect(200)
+        .then((res) => {
+          expect(res.text.includes('data-test-id="error-page-title"')).toEqual(true);
+          expect(res.text.includes('Not authorised')).toEqual(true);
+        });
+    });
+
+    it('should return the correct status and text if response.success is true', async () => {
+      viewUserContext.getViewUserContext = jest.fn()
+        .mockImplementation(() => Promise.resolve(mockAddUserData));
+      addUserContext.postUserStatus = jest.fn()
```

Comment:
I don't think this is the right mock? We are using a different controller to enable and disable users. Not the add one. Is that right?

Reply:
I've tidied them all up :)